### ISSUE TITLE
col.run_spectrum_int fails when bias tee is on?

### ISSUE THREAD

**Issue by EmmanuelSchaan** [evanmayer/rtlobs#2]
TLDR: the example code works everytime when the bias tee is off, but it fails most of the times when the bias tee is on.

Thank you so much for sharing this code, it is amazing! I am sure my issue is something I am doing wrong, but I can't seem to figure out what it is. I have a wifi antenna connected to a LNA (Sawbird+ H1), connected to a RTL SDR v3 dongle, plugged into a raspberry pi 4. I have no trouble running the example code when the bias tee is off:
![code_working](https://user-images.githubusercontent.com/2760268/236991220-298a3a2a-cc10-4a41-b8a5-72c12844203f.png)
which gives me this success message:
![result_working](https://user-images.githubusercontent.com/2760268/236991259-f690e0bb-012c-492d-83ca-49b73c64dd24.png)
and produces this cool plot:
![plot_working](https://user-images.githubusercontent.com/2760268/236991284-109afaa6-4a52-43b0-b8f2-b6ca3462cce6.png)
I can do ten times in a row, and it works every time.

However, as soon as I turn on the bias tee on the RTL SDR v3 dongle, either with the shell command:
rtl_biast -b 1
or using this code:
![code_not_working](https://user-images.githubusercontent.com/2760268/236991353-419058ec-1e36-45f8-8278-278124bfee35.png)
I can see that the bias tee is on (a LED turns on on my LNA), but I get the following error message:
![result_not_working](https://user-images.githubusercontent.com/2760268/236991382-9fcc71fb-c4c6-4e81-b31e-5938b5d68c43.png)
If I rerun the python script many many times, I get it to successfully run successfully maybe once every twenty attempts or so.
Do you know what I might be doing wrong? Apologies if this is obvious, or if I haven't given enough information.
Thank you!

Best,
Manu

**Comment by evanmayer** [comment]
Hi Manu, thanks for trying the code out! Sorry to hear things aren’t working. I suspect this is hardware-related. Are you familiar with the other `rtl_` command-line tools (`rtl_test`, `rtl_power`, etc.)? Using those with the bias tee on and off could rule out or implicate the Python code as having a role. If the problem persists, it might be best to take this issue to another forum. 

A few things to consider when you’re using an rpi and power-related things begin to work strangely: check the rpi power supply (should be ~5V, 2A rated for rpi4, and not too long a cable), consider the quality or length of any USB cable or hub between the rpi and the dongle, and consider if other devices are plugged into the rpi USB as well. A basic keyboard and mouse shouldn’t be an issue, hard drives or other peripherals might.

There is also the distant possibility of a manufacturing error with your board, which would be trickier to diagnose. Counterfeits of the RTL-SDRblog v3 are common, and I assume may increase the chances of a manufacturing error. Now is a good time to take a close look at the device’s case, maybe open it up and inspect the board, comparing things to the first two images on [this page](https://www.rtl-sdr.com/buy-rtl-sdr-dvb-t-dongles/).

**Comment by EmmanuelSchaan** [comment]
Hi Evan, thank you so much for your prompt and thoughtful reply! I agree with you, this is most likely a power issue on my side. Sorry for not realizing this.

When I first posted the issue on github, I hadn't realized that it sometimes worked when the bias tee is on (I thought it never worked). But having it work randomly like this, when power is supplied through USB by the RPI, certainly points to a power issue for my raspberry pi. Since then, I have found that the code works much more often when I plug the RTL SDR on the USB 2 ports of the RPI instead of the USB3 ports... which again points to some weird power issue. My RTL dongle is genuine I think (bought from a link from the webpage you linked https://www.rtl-sdr.com/buy-rtl-sdr-dvb-t-dongles/). I will try using rtl_test, rtl_power, etc, to learn more, and try different power supplies.

Thank you again for solving my issue, and for your considerate, thoughtful and prompt answer!

Best,
Emmanuel
